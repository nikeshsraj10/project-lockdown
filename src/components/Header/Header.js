import React from 'react';
import {Link} from 'react-router-dom';
import 'bootstrap/dist/js/bootstrap.min.js'; 
import classes from './Header.module.css';

const Header = (props) => (
    <React.Fragment>
        <nav className="navbar navbar-expand-md navbar-dark bg-dark">
            <Link to="/" className="navbar-brand">User List</Link>
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarsExampleDefault" aria-controls="navbarsExampleDefault" aria-expanded="false" aria-label="Toggle navigation">
            <span className="navbar-toggler-icon"></span>
            </button>
        
            <div className="collapse navbar-collapse" id="navbarsExampleDefault">
            <ul className="navbar-nav mr-auto">
                <li className="nav-item active">
                    <Link to="/" className="nav-link"><div className={classes.left}>Home</div><span className="sr-only">(current)</span></Link>
                </li>
                {/* <li className="nav-item dropdown">
                <a className="nav-link dropdown-toggle" href="#" id="dropdown01" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Dropdown</a>
                <div className="dropdown-menu" aria-labelledby="dropdown01">
                    <a className="dropdown-item" href="#">Action</a>
                    <a className="dropdown-item" href="#">Another action</a>
                    <a className="dropdown-item" href="#">Something else here</a>
                </div>
                </li> */}
            </ul>
            <form className="form-inline my-2 my-lg-0" style={{marginRight: '0px'}}>
                <input className="form-control mr-sm-2" type="text" placeholder="Search" aria-label="Search"/>
                <button className="btn btn-secondary my-2 my-sm-0" type="submit">Search</button>
            </form>
            <div id="addUserButton">
                <Link to="/createUser" className={["btn", "btn-primary", classes.left].join(' ')}>Add User</Link>
            </div>
            </div>
    </nav>
  </React.Fragment>
);

export default Header;