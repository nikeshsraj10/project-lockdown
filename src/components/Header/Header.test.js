import React from 'react';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';



afterEach(cleanup);
describe('Header Component Test Suite', () => {
    it("matches snapshot", () => {
        const tree = renderer.create(<BrowserRouter><Header/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
