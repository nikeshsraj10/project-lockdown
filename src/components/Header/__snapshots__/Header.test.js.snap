// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test Suite matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    User List
  </a>
  <button
    aria-controls="navbarsExampleDefault"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarsExampleDefault"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarsExampleDefault"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          <div
            className="left"
          >
            Home
          </div>
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
    <form
      className="form-inline my-2 my-lg-0"
      style={
        Object {
          "marginRight": "0px",
        }
      }
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2"
        placeholder="Search"
        type="text"
      />
      <button
        className="btn btn-secondary my-2 my-sm-0 mr-2"
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      id="addUserButton"
    >
      <a
        className="btn btn-primary left"
        href="/user"
        onClick={[Function]}
      >
        Add User
      </a>
    </div>
  </div>
</nav>
`;

exports[`matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    User List
  </a>
  <button
    aria-controls="navbarsExampleDefault"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarsExampleDefault"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarsExampleDefault"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          <div
            className="left"
          >
            Home
          </div>
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
    <form
      className="form-inline my-2 my-lg-0"
      style={
        Object {
          "marginRight": "0px",
        }
      }
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2"
        placeholder="Search"
        type="text"
      />
      <button
        className="btn btn-secondary my-2 my-sm-0 mr-2"
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      id="addUserButton"
    >
      <a
        className="btn btn-primary left"
        href="/user"
        onClick={[Function]}
      >
        Add User
      </a>
    </div>
  </div>
</nav>
`;
